webhook/resources/backup/mutator.go

```diff
@@ -85,6 +86,31 @@ func (b *backupMutator) Create(request *admission.Request, newObj runtime.Object
 		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/backupMode", "value": "%s"}`, string(longhorn.BackupModeIncremental)))
 	}
 
+	backupTargetName, ok := backup.Labels[types.LonghornLabelBackupTarget]
+	if !ok || backupTargetName == "" {
+		volume, err := b.ds.GetVolumeRO(volumeName)
+		if err != nil && (!datastore.ErrorIsNotFound(err) || backup.Spec.SnapshotName != "") {
+			err := errors.Wrapf(err, "failed to get the volume %v of backup %v", volumeName, backup.Name)
+			return nil, werror.NewInvalidError(err.Error(), "")
+		}
+		if volume != nil && volume.Spec.BackupTargetName != backupTargetName {
+			backupTargetName = volume.Spec.BackupTargetName
+		}
+	}
```

Comment:
If `backupTargetName` is not found in `backup.Labels[types.LonghornLabelBackupTarget]` and the volume is not found as well, should we reject the creation request?

Reply:
Modified. Thanks